launcher/JavaDownloader.cpp

```diff
@@ -0,0 +1,321 @@
+                                             "&bundle_type=jre"  // jre only
+                                             "&latest=true"      // only get the one latest entry
+                                             )
+                                         .arg(javaVersion, azulOS, arch, bitness),
+                                     metaResponse));
+    connect(downloadJob, &NetJob::finished, [downloadJob, metaResponse, this] {
+        disconnect(this, &Task::aborted, downloadJob, &NetJob::abort);
+        downloadJob->deleteLater();
+        delete metaResponse;
+    });
+    connect(this, &Task::aborted, downloadJob, &NetJob::abort);
+    connect(netJob, &NetJob::failed, this, &JavaDownloader::emitFailed);
+    connect(downloadJob, &NetJob::progress, this, &JavaDownloader::progress);
+    connect(downloadJob, &NetJob::succeeded, [metaResponse, isLegacy, this] {
+        QJsonParseError parse_error{};
+        QJsonDocument doc = QJsonDocument::fromJson(*metaResponse, &parse_error);
+        if (parse_error.error != QJsonParseError::NoError) {
+            qWarning() << "Error while parsing JSON response at " << parse_error.offset << " reason: " << parse_error.errorString();
+            qWarning() << *metaResponse;
+            return;
+        }
+        auto array = Json::ensureArray(doc.array());
+        if (!array.empty()) {
+            downloadAzulJava(isLegacy, array);
+        } else {
+            emitFailed(tr("No suitable JRE found"));
+        }
+    });
+    downloadJob->start();
+}
+void JavaDownloader::mojangOStoAzul(const QString& OS, QString& azulOS, QString& arch, QString& bitness)
+{
+    if (OS == "mac-os-arm64") {
+        // macos arm64
+        azulOS = "macos";
+        arch = "arm";
+        bitness = "64";
+    } else if (OS == "linux-arm64") {
+        // linux arm64
+        azulOS = "linux";
+        arch = "arm";
+        bitness = "64";
+    } else if (OS == "linux-arm") {
+        // linux arm (32)
+        azulOS = "linux";
+        arch = "arm";
+        bitness = "32";
+    } else if (OS == "linux") {
+        // linux x86 64 (used for debugging, should never reach here)
+        azulOS = "linux";
+        arch = "x86";
+        bitness = "64";
+    }
+}
+void JavaDownloader::downloadAzulJava(bool isLegacy, const QJsonArray& array)
+{  // JRE found ! download the zip
+    setStatus(tr("Downloading java from Azul"));
+    auto downloadURL = QUrl(array[0].toObject()["url"].toString());
+    auto download = new NetJob(QString("JRE::DownloadJava"), APPLICATION->network());
+    auto temp = std::make_unique<QTemporaryFile>(FS::PathCombine(APPLICATION->root(), "temp", "XXXXXX.zip"));
+    FS::ensureFolderPathExists(FS::PathCombine(APPLICATION->root(), "temp"));
+    // Have to open at least once to generate path
+    temp->open();
+    temp->close();
+    download->addNetAction(Net::Download::makeFile(downloadURL, temp->fileName()));
```

Comment:
why do all this for the temp file? can't we just create a simple `QTemporaryFile` in the system's temp folder and use it? Or, at the very least, use the same temporary dir we use elsewhere (i.e. `.LAUNCHER_TEMP`)